Feature - Add table view for Users

mnigh opened the pull request:
Resolves #46.

Visible via storybook.

tristan-orourke reviewed (changes requested):
The table code looks good (I love how the GlobalFilter applies to all the fields!), I just want to resolve some formatting inconsistencies.

I see a lot of styling linting errors, such as single-quotes instead of double, and tab spacing in fakeUsers.ts. Lets figure out why we have different linting configs.

I also see a linting error that starts with `Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.` at the first import statement in all the React components. I _think_ it can be resolved by changing the "include" field in tsconfig.json to "resources/js/**/*".

Finally, my environment isn't recognizing the TS types from react-table-config.d.ts. That's probably something on my end, but I want to be sure.

tristan-orourke reviewed (changes requested):
The table code looks good (I love how the GlobalFilter applies to all the fields!), I just want to resolve some formatting inconsistencies.

I see a lot of styling linting errors, such as single-quotes instead of double, and tab spacing in fakeUsers.ts. Lets figure out why we have different linting configs.

I also see a linting error that starts with `Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.` at the first import statement in all the React components. I _think_ it can be resolved by changing the "include" field in tsconfig.json to "resources/js/**/*".

Finally, my environment isn't recognizing the TS types from react-table-config.d.ts. That's probably something on my end, but I want to be sure.

mnigh commented:
> The table code looks good (I love how the GlobalFilter applies to all the fields!), I just want to resolve some formatting inconsistencies.
> 
> I see a lot of styling linting errors, such as single-quotes instead of double, and tab spacing in fakeUsers.ts. Lets figure out why we have different linting configs.
> 
> I also see a linting error that starts with `Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.` at the first import statement in all the React components. I _think_ it can be resolved by changing the "include" field in tsconfig.json to "resources/js/**/*".
> 
> Finally, my environment isn't recognizing the TS types from react-table-config.d.ts. That's probably something on my end, but I want to be sure.

I've updated some of the formatting and syntax, please let me know if I've missed anything, thanks. Also, I think the types package for react-table that I added and then removed is required. I've added it again!
